### PR TITLE
Update explorer.core CLI, update READMEs

### LINKED ISSUE #97: Make `core` CLIs UX uniform
Currently, the two CLIs `explorer.core` and `explorer.extract` in package `core` have different UX.

```
$ poetry run python -m explorer.core --help
Usage: python -m explorer.core [OPTIONS]

Options:
  --out PATH              The output path at which to write the JSON Schema
  --stdout                Write outputs to stdout instead of to a file
  --pretty / --no-pretty  Enable/disable pretty printing of the output JSON
                          Schema
  --help                  Show this message and exit.

$ poetry run python -m explorer.extract --help
Usage: python -m explorer.extract [OPTIONS] OUTFILE

  Extract the graph of derivations from a flake as JSON.

  OUTFILE is the path to the output file to write to, use "-" to write to
  stdout.

Options:
  --target-flake-ref TEXT  The reference of the target Nix Flake
  --target-system TEXT     The system in which to evaluate the packages
  -v, --verbose            Increase verbosity
  --help                   Show this message and exit.
```

Also, `explorer.core` should be more POSIX compliant.

Towards this goal, in `explorer.core`:
- `--out PATH`
  - should instead be a positional argument
  - should take `-` for `stdout`
- `--stdout` should be removed

Once updated, the doc should be updated as well.

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
Closes #97 and updates READMEs according to #116

### CODE REVIEW

File: core/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ poetry install --only main
 To extract the data from nixpkgs, use:
 
 ```console
-poetry run python -m explorer.extract graph.json
+poetry run python -m explorer.extract derivations.json
```
Comment (zz1874) [inline core/README.md:22]:
```suggestion
poetry run python -m explorer.extract derivations.jsonl
```

**Review by zz1874 — CHANGES_REQUESTED** [review]
Just a minimal suggestion. Others look good to me.

File: core/explorer/extract/cli.py [hunk 2]
```diff
@@ -263,17 +267,13 @@ def try_queue_get():
...
+    outfile: IO[str],
     target_flake_ref: str,
     target_system: str,
     n_workers: int,
     offline: bool,
     verbose: bool,
-    outfile: IO[str],
 ):
     """
     Extract the graph of derivations from a flake as JSON.
```
Comment (GuillaumeDesforges) [inline core/explorer/extract/cli.py:279]:
```suggestion
    Extract the graph of derivations from a flake as JSONL.
```

File: api/README.md [hunk 3]
```diff
@@ -12,11 +12,17 @@ poetry install --only main
 
 ### Load data
 
-This package comes with a command line utility called `explorer-ingest-graph` to load a Nix graph JSON file that follows [`nixpkgs-graph.schema.json`](../core/nixpkgs-graph.schema.json).
+This package comes with a command line utility to load a JSONL file of Nix derivations, where each line follows [`derivation.schema.json`](../core/derivation.schema.json).
 To extract such JSON, see [the `core` package instructions](../core/README.md).
```
Comment (GuillaumeDesforges) [inline api/README.md:16]:
```suggestion
To extract such JSONL, see [the `core` package instructions](../core/README.md).
```

**Pull request merged**